### PR TITLE
Support pushing images to other container registries

### PR DESCRIPTION
**Pull request by paulfantom**
Signed-off-by: paulfantom <pawel@krupa.net.pl>

* add easy way to add new registries (just add their prefix to `REGISTRIES` var) 
* push images to github container registry
* ability to run locally
* better tagging of development images

**Release Note Template (will be copied)**

```release-note:enhancement
Added publishing images on GitHub Container Registry
```

### CODE REVIEW

**Comment by paulfantom** [comment]
Closes #3258
Closes #3234

**Comment by paulfantom** [comment]
I would like to merge this before 0.49.0 (I am a release shepherd for this version).

If/When this is merged I will enable container registry for prometheus-operator.

/cc @prometheus-operator/prometheus-operator-reviewers

File: Makefile
Comment (paulfantom) [inline Makefile:14]: Lint image is not used anywhere :shrug: 

For other images, I changed the variable names for clarity.

File: scripts/push-docker-image.sh
Comment (paulfantom) [inline scripts/push-docker-image.sh:7]: `GITHUB_REF` is not set when using the script locally, hence I deleted this check.

File: scripts/push-docker-image.sh
Comment (paulfantom) [inline scripts/push-docker-image.sh:30]: Does anyone actually use those `-dev` images? We don't list them anywhere in any docs so discovery is a bit hard. Also, those are build only from `master` branch and branches in the repo, so not really helping during standard development workflow. Maybe we should just remove them? @prometheus-operator/prometheus-operator-reviewers WDYT?

Comment (simonpasquier) [reply scripts/push-docker-image.sh:30]: No strong opinion but I can see how it can be helpful when you want to have users testing a bug fix or a change without having them to rebuild locally.

Comment (paulfantom) [reply scripts/push-docker-image.sh:30]: I would agree if those images were built from PRs. As it is right now, the maintainer would need to create a branch to test the bug fix or a change and wait for CI to publish images. Personally, I would be in favor of simplifying image building process than keeping those `-dev` images.

Comment (simonpasquier) [reply scripts/push-docker-image.sh:30]: Fair enough. If nobody had ever used this possibility, I agree that it's not worth keeping it.

File: Makefile
Comment (simonpasquier) [reply Makefile:14]: should we remove `po-lint` completely?

Comment (paulfantom) [reply Makefile:14]: I do believe there are better tools for the job (like `kubeconform`) and we should point to those instead of maintaining our own tool for the same thing. Plus the same feature is actively being implemented in `kubectl`.

However, let's do that in a separate PR (here is an issue: https://github.com/prometheus-operator/prometheus-operator/issues/4099). I'll leave this line here though as it is dead code either way.

**Review by simonpasquier — APPROVED** [review]
lgtm

**Pull request merged**
